L/libcxxwrap_julia/build_tarballs.jl

```diff
@@ -18,23 +19,36 @@ sources = [
 script = raw"""
 mkdir build
 cd build
-cmake -DJulia_PREFIX=$prefix -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_FIND_ROOT_PATH=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release ../libcxxwrap-julia/
+
+# workaround for 32bit intel builds, where Julia's atomics.h will otherwise
+# complain
+if [[ "${target}" == i686* ]]; then
+    export CXXFLAGS="-msse -msse2"
```

Comment:
I really don't like this bit. I feel that it should be up to libjulia_jll to ensure that projects using it gets suitable compiler flags, somehow. As it is, we'll likely have to put this into all JLL build recipes which are using libjulia_jll, and that doesn't scale well.

How to do that? Perhaps teach `julia-config.jl --cflags` (by patching it) to add these? But is libcxxwrap even using this? If not, perhaps it should?

Alternatively, perhaps libcxxwrap_julia could extend its CMake macros to insert these compilers flags, then at least it and presumably all (?) JLLs using it would be OK?

Reply:
I hadn't found the time to investigate this, but it looks weird that we have to specify those features, since those should be default for the target platform

Reply:
```
        "i686" => Dict(
            "i686" => ["-march=i686", "-mtune=generic"],
            "prescott" => ["-march=prescott", "-mtune=prescott"],
        ),
```
And here `i686` is used which doesn't enable SSE by default. That would be the case with `prescott`, though. 

So I am going out on a limb here, as admittedly I have not understood this part of BB at all, but: Perhaps the problem then is that libjulia_jll is built for i686 instead of prescott? I think BB tries to "guess" this, but for the x86_64 builds of libjulia, I see this warning:
> ##[warning]libjulia.so.1.6 contains a `cpuid` instruction; refusing to analyze for minimum instruction set, as it may dynamically select the proper instruction set internally.  Would have chosen avx2, instead choosing x86_64.

Now to my surprise I see no similar warning for the i686 builds? Could it perhaps be tha the code detecting the minimal march is incomplete and/or buggy?

Reply:
Aha, from `minimum_march` in `BinaryBuilder.jl`:
```
    elseif arch(p) == "i686"
        if counts["sse3"] > 0
            return "prescott"
        end
```
But the only alternative right now is to use `i686` as arch, which means pentium pro. To get SSE and SSE2, you'd want at least pentium4.

So either another arch ("pentium4") needs to be added, or else the prescott check be extended to also cover MMX, SSE and SSE2.

Reply:
Third option: changing `i686` to actually map to `pentium4` (given that Julia won't run on anything older, so why bother restricting JLL binaries further than that? Might as well benefit from additional optimizations, too.